drivers/display/oled/oled.c

```diff
@@ -0,0 +1,320 @@
+ *        row
+ *        column
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t move_cursor(struct ssd_1306_dev *device, uint8_t row, uint8_t column)
+{
+	int32_t ret = SUCCESS;
+	uint8_t command[3];
+	ret |= gpio_set_value(device->dc_pin, DC_CMD);
+
+	command[0] = 0x21;
+	command[1] = column*8;
+	command[2] = COL_NB * 8 - 1U;
+	ret |= spi_write_and_read(device->spi_desc, &command, 3U);
+
+	command[0] = 0x22;
+	command[1] = row;
+	command[2] = ROWS_NB - 1U;
+	ret |= spi_write_and_read(device->spi_desc, &command, 3U);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Clears data on display.
+ *
+ * @param device - The device structure.
+ *        row    - numbers of rows
+ *        column - numbers of columns
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_clear(struct ssd_1306_dev *device, uint8_t rows, uint8_t columns)
+{
+	int32_t ret = SUCCESS;
+	uint8_t i, j;
+
+	ret |= gpio_set_value(device->dc_pin, DC_DATA);
+
+	for(i=0; i<rows; i++)
+		for(j=0; j<columns; j++)
+			ret |= display_print_ascii(device, ' ', i, j);
+
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Prints character at selected position.
+ *
+ * @param device - The device structure.
+ *        row    - row
+ *        column - column
+ *        ascii  - corresponding number to ascii table
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_print_ascii(struct ssd_1306_dev *device, uint8_t ascii, uint8_t row, uint8_t column)
+{
+	int32_t ret = SUCCESS;
+	int8_t ch[8];
+	int8_t i;
+
+	ret |= move_cursor(device, row, column);
+
+	ret |= gpio_set_value(device->dc_pin, DC_DATA);
+
+	for(i=0; i<8; i++) //LSB at tx_char[8]
+	{
+		ch[7-i] = (chr_map[ascii] >> (8*i)) & 0xFF;
+	}
+
+	ret |= spi_write_and_read(device->spi_desc, ch, 8U);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Prints char string at selected position.
+ *
+ * @param device - The device structure.
+ *        row    - numbers of rows
+ *        column - numbers of columns
+ *        *msg   - char string pointer
+ *
+ * @return SUCCESS in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t display_print_string(struct ssd_1306_dev *device, char *msg, uint8_t row, uint8_t column)
+{
+	int32_t ret = SUCCESS;
+	int32_t len;
+	int32_t i;
+	int32_t j;
+	int32_t r = row;
+	int32_t c = column;
+
+	len = strlen(msg);
+	for(i=0; i<len; i++){
+		if(r<ROWS_NB){
```

Comment:
nitpick: spacings are required here:
`if(r<ROWS_NB){` -> `if (r < ROWS_NB) {`

Check the entire code for similar issues.